debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/UniqueDatabase.java

```diff
@@ -193,6 +193,10 @@ public Configuration.Builder defaultJdbcConfigBuilder() {
         Builder builder = Configuration.create()
                 .with(MySqlConnectorConfig.HOSTNAME, System.getProperty("database.hostname", "localhost"))
                 .with(MySqlConnectorConfig.PORT, System.getProperty("database.port", "3306"))
+                .with(MySqlConnectorConfig.JDBC_PROTOCOL, System.getProperty("database.protocol",
+                        MySqlConnectorConfig.JDBC_PROTOCOL.defaultValueAsString()))
+                .with(MySqlConnectorConfig.JDBC_DRIVER, System.getProperty("database.jdbc.driver",
+                        MySqlConnectorConfig.JDBC_DRIVER.defaultValueAsString()))
```

Comment:
Wouldn't it make it easier for users when they have a single switch that sets the backend to on of the supported MySQL types, like mysql, percona, mariadb, etc and we set related parameters accordingly instead of providing that full detail level?

Reply:
Yes, there is a Jira to introduce a strategy pattern for these use cases.  But rather than shoehorn that into the code, we're doing this piecemeal so we see all the touch points we need to refactor more clearly over the 2.5 release cycle.